lib/Controller/ApplicationController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @author Baptiste Fotia <baptiste.fotia@arawa.fr> for Arawa (https://arawa.fr)
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\GroupFolders\Controller;
+
+use OCA\GroupFolders\Service\ApplicationService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\JSONResponse;
+
+class ApplicationController extends Controller {
+
+    /** @var ApplicationService */
+    private $applicationService;
+    
+    public function __construct(
+        ApplicationService $applicationService
+    )
+    {
+        $this->applicationService = $applicationService;
+    }
+
+    /**
+     * @NoAdminRequired
+     * @NoCSRFRequired
+     * @return JSONResponse
+     */
+    public function checkAppsBasedOnGroupfolders() {
```

Comment:
I prefer if this was given to the setting ui with IInitialState

Reply:
Hmmm I'm sorry, I don't understand. Do you want this function has a parameter to check if apps are installed ?
If that's what it is, we might as well do it on the backend

What is "IInitialState" ? It is a function ? Back-End or Front-End ?

Reply:
See for example apps/federatedfilesharing/lib/Settings/Admin.php in server, IInitalState allow to directly provide some value to the frontend, that can then be loaded with loadState(appName, key)

Reply:
Okay, I try it this morning !

But, I don't understand how to share the data between the back-end and the front-end. Does the IInitialState save the data to the database or to a file ?

Reply:
It... It... It works ! What is this black magic ?! :eyes: :magic_wand: 
Where is this data saved ? :eyes: 

Currently, here are my change :

```php
<?php
//...
// Admin.php
class Admin implements IDelegatedSettings {


	public function __construct(IInitialState $initialState,
		ApplicationService $applicationService,
	)
	{
		$this->initialState = $initialState;
		$this->applicationService = $applicationService;
	}
	
	public function getForm(): TemplateResponse {

		$this->initialState->provideInitialState('checkAppsInstalled', $this->applicationService->checkAppsInstalled());
		
		
		return new TemplateResponse(
			'groupfolders',
			'index',
			['appId' => 'groupfolders'],
			''
		);
	}
//...
```

```typescript
// App.tsx
export class App extends Component<{}, AppState> implements OC.Plugin<OC.Search.Core> {
       //...
	componentDidMount() {
		this.api.listFolders().then((folders) => {
			this.setState({folders});
		});
		this.api.listGroups().then((groups) => {
			this.setState({groups});
		});
		this.api.isAdminNextcloud().then((isAdminNextcloud) => {
			this.setState({isAdminNextcloud});
		});
		this.setState({ checkAppsInstalled: loadState('groupfolders', 'checkAppsInstalled') })
		// this.api.checkAppsBasedOnGroupfolders().then((checkAppsInstalled) => {
		// 	this.setState({checkAppsInstalled})
		// });
		OC.Plugins.register('OCA.Search.Core', this);
	}
//...
}
```

Reply:
I think I resolved this fix. Please, can you review my code ? :pray: 

Here is the commit for change : https://github.com/nextcloud/groupfolders/pull/2072/commits/0a3dddb1bf407b57143085316141f0f65067cc65